### DISCUSSION TITLE
Tutorial example for history access yields error

### DISCUSSION THREAD

**Discussion by tsobczynski** [xonsh/xonsh#4819] [Q&A]
In the Tutorial page [Bash to Xonsh Translation Guide](https://xon.sh/bash_to_xsh.html), there is an entry with the note "Get the last argument of the last command". With a fresh installation of Xonsh in two different environments, I see an error if I try out the Xonsh version:

```
>>> __xonsh__.history[-1, -1]
xonsh: For full traceback set: $XONSH_SHOW_TRACEBACK = True
TypeError: history indices must be integers or slices, not <class 'tuple'>
```

I get a similar error result with the first example under History Indexing on the [History](https://xon.sh/tutorial_hist.html) Tutorial page:

```
>>> echo mkdir with/a/huge/name/
mkdir with/a/huge/name/
>>> __xonsh__.history[-1, -1]
xonsh: For full traceback set: $XONSH_SHOW_TRACEBACK = True
TypeError: history indices must be integers or slices, not <class 'tuple'>
```

If I look at what is actually in this structure, it doesn't look like the history is preserved as a list of tokens that could be sliced:

```
>>> echo mkdir with/a/huge/name/
mkdir with/a/huge/name/
>>> __xonsh__.history[-1].cmd
'echo mkdir with/a/huge/name/\n'
```

Is this just my mistake? Is the documentation in error? What is the recommended way to achieve this action (get last argument of last command)?

**Comment by jnoortheen** [comment]
It could be caused by any regression. As you have mentioned, the cmd is stored as string. You can check the showcmd alias source code to create a similar alias to parse and return the cmd args. Other option would be to use the '__xonsh__.execer.parse'
